### PR TITLE
Add doc comment about potential of events to arrive out of order.

### PR DESCRIPTION
**Pull request by PeterAdams-A**
Motivation:

This has come as a surprise to at least one person and is easy to forget.

Modifications:

Doc commented added at the top of ChannelPipeline.

Result:

Docs are clearer.

### CODE REVIEW

File: Sources/NIO/ChannelPipeline.swift [hunk 1]
```diff
@@ -109,6 +109,11 @@
 /// - 1 and 2 don't implement `ChannelOutboundHandler`, and therefore the actual evaluation order of a outbound event will be: 5, 4, and 3.
 /// - If 5 implements both `ChannelInboundHandler` and `ChannelOutboundHandler`, the evaluation order of an inbound and a outbound event could be 125 and 543 respectively.
 ///
+/// Note:   Handlers may choose not to progagate messages down the pipeline immediaitely.  For example a handler may need to wait
```
Comment (Lukasa) [inline Sources/NIO/ChannelPipeline.swift:112]:
```suggestion
/// Note:   Handlers may choose not to progagate messages down the pipeline immediately.  For example a handler may need to wait
```

Comment (yim-lee) [reply Sources/NIO/ChannelPipeline.swift:112]: In addition to Cory's suggestion:

```suggestion
/// Note:   Handlers may choose not to propagate messages down the pipeline immediaitely.  For example a handler may need to wait
```

Comment (ktoso) [reply Sources/NIO/ChannelPipeline.swift:112]: This reads good, thanks @PeterAdams-A !

The spacing is a bit weird... Why double spaces everywhere?

```suggestion
Note: Handlers may choose not to progagate messages down the pipeline immediately. For example a handler may need to wait
```

etc?

**Comment by ktoso** [comment]
Resolves https://github.com/apple/swift-nio/issues/1607

**Comment by PeterAdams-A** [comment]
Interestingly commit comments seem to disappear when multiple people suggest changes to the same line.  I've corrected both typos though.

The multiple space thing comes from being old - it's what we used to do.  I was going to accept the changes but then I noticed that the comments above this area formatted the same way so I'll leave for now so at least the file is consistent with itself.

**Pull request merged**